inttest/capi-docker/capi_docker_test.go

```diff
@@ -0,0 +1,203 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package capidocker
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/k0sproject/k0smotron/inttest/util"
+	"github.com/stretchr/testify/suite"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/k0sproject/k0s/inttest/common"
+	k0stestutil "github.com/k0sproject/k0s/inttest/common"
+)
+
+type CAPIDockerSuite struct {
+	suite.Suite
+	client           *kubernetes.Clientset
+	restConfig       *rest.Config
+	clusterYamlsPath string
+}
+
+func TestCAPIDockerSuite(t *testing.T) {
+	s := CAPIDockerSuite{}
+	suite.Run(t, &s)
+}
+
+func (s *CAPIDockerSuite) SetupSuite() {
+	kubeConfigPath := os.Getenv("KUBECONFIG")
+	s.Require().NotEmpty(kubeConfigPath)
+	// Get kube client from kubeconfig
+	restCfg, err := clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+	// t.Logf("restCfg.Host: %s", restCfg.Host)
+	// t.Logf("restCfg full:\n%+v", restCfg)
+	s.Require().NoError(err)
+	s.Require().NotNil(restCfg)
+	s.restConfig = restCfg
+
+	// Get kube client from kubeconfig
+	kubeClient, err := kubernetes.NewForConfig(restCfg)
+	s.Require().NoError(err)
+	s.Require().NotNil(kubeClient)
+	s.client = kubeClient
+
+	tmpDir := s.T().TempDir()
+	s.clusterYamlsPath = tmpDir + "/cluster.yaml"
+	s.Require().NoError(os.WriteFile(s.clusterYamlsPath, []byte(dockerClusterYaml), 0644))
+}
+
+func (s *CAPIDockerSuite) TestCAPIDocker() {
+
+	// Dump the cluster yaml to a file
+	tmpDir := s.T().TempDir()
+	tmpFile := tmpDir + "/cluster.yaml"
+	s.Require().NoError(os.WriteFile(tmpFile, []byte(dockerClusterYaml), 0644))
```

Comment:
Do we need it both in `SetupSuite` and in `TestCAPIDocker`?

Reply:
Absolutely not. The test func version is a left-over. I'll remove it